src/app.py

```diff
@@ -783,11 +821,40 @@ def app_manifest(app):
 
     manifest, extracted_app_folder = _extract_app(app)
 
-    shutil.rmtree(extracted_app_folder)
-
     raw_questions = manifest.get("install", {}).values()
     manifest["install"] = hydrate_questions_with_choices(raw_questions)
 
+    # Add a base64 image to be displayed in web-admin
+    if Moulinette.interface.type == "api":
+        import base64
+
```

Comment:
Sooo I was thinking about maybe having another route `GET /app/screenshots/whatever` to fetch screenshot asynchronously on the webadmin, otherwise this may be kinda slow on bad internet connection ... but after thinking about it, this would probably mean that YunoHost would download the app *twice* unless we implement some smarter cache system :|

I did try to add a check in the package linter to recommend to limit the size of the screenshots/ folder to 512k, though this was only an "Info" and I see quite a lot of apps with kinda large files. I'm switching this to a Warning now ... + add a check on allowed file types

I would still add a `with_screenshot` option to `app_manifest()` though, because it's likely to be called by other functions in the code and you don't want to spend time encoding base64 screenshots for these, that's only useful for when the webadmin displays the install form / catalog view. Watcha think ? :grimacing:

Reply:
Added `with_screenshot` in actionmap!